sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,867 @@
+
+[NOTE]
+====
+Unlike the functions defined in the first four tables, the functions defined in this section do not correspond to any functions in the core SYCL specification.
+====
+
+[[table.usm.additional.allocs]]
+.USM Allocation Functions Requiring usm_alloc
+[width="100%",options="header",separator="@",cols="65%,35%"]
+|====
+@ Function @ Description
+a@
+[source]
+----
+template <typename propertyListA, typename propertyListB>
+sycl::ext::oneapi::experimental::annotated_ptr<void, propertyListB>
+sycl::ext::oneapi::experimental::malloc(size_t numBytes,
+                                        const device& syclDevice,
+                                        const context& syclContext,
+                                        const propertyListA &propList)
+----
+a@
+The propList must contain a `usm_alloc` property, otherwise the compiler will issue a diagnostic error.
+Returns a [code]#usm_alloc# allocation.
+This allocation is specified in bytes. This memory
+must be deallocated with [code]#sycl::free# to avoid memory leaks.
+On failure, the raw pointer of the returned annotated_ptr will be [code]#nullptr#.
+Zero or more runtime and compile-time constant properties can be provided to the allocation function
+via an instance of [code]#sycl::ext::oneapi::experimental::properties#.
+The compile-time constant properties in propList will also appear as properties of the returned annotated_ptr.
+Runtime properties in propList will not appear as properties of the returned annotated_ptr.
+The [code]#syclDevice# parameter is
+ignored if [code]#usm_alloc# is [code]#host#.  If [code]#usm_alloc# is not
+[code]#host#, [code]#syclDevice# must either be contained by
+[code]#syclContext# or it must be a descendent device of some device that
+is contained by that context, otherwise this function throws a synchronous
+[code]#exception# with the [code]#errc::invalid# error code.
+
+Available only if `propertyListA` and `propertyListB` are specializations of the sycl::ext::oneapi::experimental::properties class.
+
+|====
+
+=== USM Memory Allocation Properties 
+
+This section specifies the properties that can be passed to the USM memory allocation functions defined in Section <<section.usm.allocs>>.
+A SYCL implementation or SYCL backend may support additional properties other than those defined here, provided they are defined in accordance with the
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties] extension.
+If any other properties are passed to the USM memory allocation functions with `properties` support, the compiler will issue a diagnostic error.
+
+All properties that are valid for annotated_ptr must be supported by the USM memory allocation functions with `properties` support.
+Unless otherwise specified, supported properties do not affect the behavior of the USM memory allocation functions with `properties` support.
+
+[NOTE]
```

Comment:
This note repeats the same information said immediately before it.  You probably only need to say it once.

Reply:
I tried to reword this so it is less repetitive. The NOTE now just mentions that the motivation is for user convenience in only needing to specify the properties in one place.